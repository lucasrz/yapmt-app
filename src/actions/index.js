import {
  SET_PROJECTS,
  SET_PROJECT,
  GET_PROJECTS_SAGA,
  CREATE_PROJECT_SAGA,
  DELETE_PROJECT_SAGA,
  ADD_TASK_SAGA,
  SET_TASK,
  UPDATE_TASK_SAGA,
  UPDATE_PROJECT_STATUS,
  SET_TASKS,
} from '../constants';

export function setProjects(projects) {
  return {
    type: SET_PROJECTS,
    projects
  };
}

export function setProject(project) {
  return {
    type: SET_PROJECT,
    project
  };
}

export function updateProjectStatus(project) {
  return {
    type: UPDATE_PROJECT_STATUS,
    status: project.status
  };
}

export function setTask(task) {
  return {
    type: SET_TASK,
    task
  };
}

export function setTasks(state) {
  return {
    type: SET_TASKS,
    tasks: state.tasks
  };
}


export function getProjectsSaga() {
  return {
    type: GET_PROJECTS_SAGA,
  };
}

export function createProjectSaga(project) {
  return {
    type: CREATE_PROJECT_SAGA,
    project
  };
}

export function deleteProjectsSaga(project) {
  return {
    type: DELETE_PROJECT_SAGA,
    project
  };
}

export function addTaskSaga(task) {
  return {
    type: ADD_TASK_SAGA,
    task
  };
}

export function updateTaskSaga(task) {
  return {
    type: UPDATE_TASK_SAGA,
    task
  };
}
