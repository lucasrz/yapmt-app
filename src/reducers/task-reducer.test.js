import reducer from './index';
import { setTask } from '../actions';

describe('Reducers', () => {
  const task = {
    isDone: true,
    _id: '5d1fff233bbd50c6e5547c3f',
    description: "aaa",
    owner: "@aaaa",
    dueDate: "11/11",
  }

  it('Set Task', () => {
    const initialState = {};
    const next = reducer(initialState, setTask(task))
    expect(next).toMatchSnapshot()
  });
});