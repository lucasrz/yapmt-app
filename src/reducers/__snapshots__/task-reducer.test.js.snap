// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducers Set Task 1`] = `
Object {
  "projectsReducer": Object {
    "projects": Array [],
  },
  "tasksReducer": Object {
    "task": Object {
      "task": Object {
        "_id": "5d1fff233bbd50c6e5547c3f",
        "description": "aaa",
        "dueDate": "11/11",
        "isDone": true,
        "owner": "@aaaa",
      },
      "type": "SET_TASK",
    },
  },
}
`;
